### PR TITLE
[aik-mod] New AIK Inline Mod plugin

### PR DESCRIPTION
**Pull request by Fryuni**

### CODE REVIEW

File: packages/aik-mod/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	};
+}
+
+type InlineModPlugin = Plugin<
+	'inlineModule',
+	'astro:config:setup',
+	(p: HookParams) => (options: ModuleOptions) => string
+>;
+
+export const inlineModPlugin: InlineModPlugin = definePlugin({
```
Comment (florian-lefebvre) [inline packages/aik-mod/src/index.ts:59]:
```suggestion
export const inlineModPlugin = definePlugin({
```
It should be inferred

File: packages/aik-mod/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	name: 'inlineModule',
+	hook: 'astro:config:setup',
+	implementation: ({ config, updateConfig }) => {
+		const ensurePlugin = ensurePluginIsInstalled({ config, updateConfig });
```
Comment (florian-lefebvre) [inline packages/aik-mod/src/index.ts:63]: Not sure if anything should be done in this function call, haven't tested myself. Out of curiosity, why here and not below?

File: packages/aik-mod/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	},
+});
+
+type DefineModPlugin = Plugin<
+	'defineModule',
+	'astro:config:setup',
+	(p: HookParams) => (name: string, options: ModuleOptions) => void
+>;
+
+export const defineModPlugin: DefineModPlugin = definePlugin({
```
Comment (florian-lefebvre) [inline packages/aik-mod/src/index.ts:77]:
```suggestion
export const defineModPlugin = definePlugin({
```
Should be inferred as well

File: packages/aik-mod/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	},
+});
+
+type DefineMiddlewarePlugin = Plugin<
+	'defineMiddleware',
+	'astro:config:setup',
+	(p: HookParams) => (order: 'pre' | 'post', handler: MiddlewareHandler) => void
+>;
+
+export const defineMiddlewarePlugin: DefineMiddlewarePlugin = definePlugin({
```
Comment (florian-lefebvre) [inline packages/aik-mod/src/index.ts:102]:
```suggestion
export const defineMiddlewarePlugin = definePlugin({
```
same

File: packages/aik-mod/src/index.ts (hunk 1 above)
Comment (Fryuni) [reply packages/aik-mod/src/index.ts:59]: https://github.com/florian-lefebvre/astro-integration-kit/issues/50#issue-2128954624 😁

File: packages/aik-mod/src/index.ts (hunk 2 above)
Comment (Fryuni) [reply packages/aik-mod/src/index.ts:63]: If the same integration declares more than one module the Vite plugin should be only added once.

Calling `updateConfig` doesn't update the `config` in the same integration, only for the next integration. So it creates a function that can be called multiple times but only adds the plugin the first time. Or in the case the plugin is already there it does nothing.

**Pull request merged**